### ISSUE TITLE
Move REST API endpoints to controllers

### ISSUE THREAD

**Issue by JakePT** [10up/ElasticPress#3648]
### Is your enhancement related to a problem? Please describe.

Following #3643 we should move our other REST API endpoints into their own controllers, for better separation of concerns.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JakePT** [comment]
To discuss:

- Whether we should instantiate the API endpoints in their Features, as they are currently, or whether they should be instantiated from the main plugin file. If the latter, what's the best method to do that?
- Whether we can rename or split some of the endpoints, eg. splitting the Facet endpoints into separate taxonomy/meta endpoints/controllers, and whether the related posts endpoint should be moved to the ElasticPress namespace.

**Comment by JakePT** [comment]
Proceed with putting controllers in REST directory. Rename some endpoints to be feature-agnostic, keep old routes for backwards compatibility.

**Comment by JakePT** [comment]
@felipeelia I’m trying to resolve failing tests with the PR #3650, but I’m getting an unexpected error when syncing. Syncing is failing because there’s an error output in the response:
```
Warning: Undefined array key "properties" in /var/www/html/wp-content/plugins/elasticpress/includes/classes/Indexable.php on line 1208
```
The relevant line is here: https://github.com/10up/ElasticPress/blob/cfced1019ff8807e06969c3cd90f1b39fc254280/includes/classes/Indexable.php#L1208 

But I don’t see anything in the PR that would mess with that: https://github.com/10up/ElasticPress/pull/3650/files
Do you have any idea what the problem might be? I can’t replicate it on the 5.0.0 branch.

**Comment by felipeelia** [comment]
@JakePT That happens because `ElasticPress\REST\MetaRange::get_args()` is being called on *every* load, because WP needs the args parameter to register the REST route. That will need to change anyway, as we really don't want to call Elasticsearch on every page load.

ps.: Adding `error_log( wp_debug_backtrace_summary() );` to that method helped me understand why that was being called in the first place.

**Comment by felipeelia** [comment]
Closed by #3650
